proof/mint.go

```diff
@@ -158,6 +158,23 @@ func (p *BaseProofParams) HaveInclusionProof(anchorOutputIndex uint32) bool {
 	return p.OutputIndex == int(anchorOutputIndex)
 }
 
+// HaveSTXOInclusionProof returns true if the inclusion proof is for the given
+// anchor output index.
+func (p *BaseProofParams) HaveSTXOExclusionProof(
+	anchorOutputIndex uint32) bool {
+
+	for _, proof := range p.ExclusionProofs {
+		if proof.OutputIndex == anchorOutputIndex &&
+			proof.CommitmentProof != nil &&
+			len(proof.CommitmentProof.STXOProofs) > 0 {
```

Comment:
How are we distinguishing between the inclusion and exclusion proofs? Don't we need to maps here? 

Also this might be a chance to clean up some of the structure of the proofs. Eg: being more explicit about when something is an inclusion proof or exclusion proof, etc.

Reply:
Or is the idea that for a given output all the prevIDs are unique as their based on outpoints? 

I think we should try to be as explicit as possible with the logic here to avoid inadvertent bugs.

Reply:
This func only checks for _exclusion_ proofs and does so by cycling through the proofs in the `ExclusionProofs` field of the `BaseProofParams`. We decided to store stxo proofs in the `commitmentProof` struct as a nice solution for not having all sorts of `UnkownOddType`s, which was the initial approach JHB and I took. That does mean that the context informs whether we are working with an in- or exclusion proof.

We do rely on `prevID`s being unique, but I fail to see how that affects the code at the point where you placed this comment. See my other comment on `prevID`s here: https://github.com/lightninglabs/taproot-assets/pull/1453#discussion_r2024950538